lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -388,6 +394,33 @@ def configure(conf)
       @resource.labels.freeze
       @common_labels.freeze
 
+      # If prometheus monitoring is enabled, register metrics in the default
+      # registry and store metric objects for future use. If metrics are
+      # registed already, just store the metric objects.
+      if @prometheus_monitoring_enabled
+        require 'prometheus/client'
+
+        prometheus = Prometheus::Client.registry
+
+        # Exception-driven behavior to avoid synchronization errors
+        begin
+          @prometheus_http_requests_count = prometheus.counter(
```

Comment:
This is generally useful beyond Prometheus. How about creating a `monitoring` module, which uses the prometheus client if available, and changing these to `monitoring.counter` instead. Something like this:
```ruby
module Monitoring
  begin
    require 'prometheus/client'
    @prometheus = Prometheus::Client.registry
  rescue LoadError
    @prometheus = nil
  end
  class Counter
    def initialize(name, desc)
      @name = name
      @desc = desc
      @map = {}
      @map.default = 0
    end
    def increment(labels, inc=1)
      @map[labels] = @map[labels] + inc
    end
  end
  def counter(name, desc)
    return @prometheus ? @prometheus.counter(name, desc) : Counter.new(name, desc)
  end
end
```
Then these variables become just `@http_requests_count` (or even just `@requests_count`) and `@log_entries_count`, and the option becomes just `@monitoring_enabled`...

We'd probably want to test the prometheus path end-to-end anyway, though.